lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+    $c->render(text => 'Hello World!');
+  };
+
+  # PUT /hello
+  put '/hello' => sub ($c) {
+    my $size = length $c->req->body;
+    $c->render(text => "You uploaded $size bytes to /hello.");
+  };
+
+  # GET|POST|PATCH /bye
+  any ['GET', 'POST', 'PATCH'] => '/bye' => sub ($c) {
+    $c->render(text => 'Bye World!');
+  };
+
+  # * /whatever
+  any '/whatever' => sub ($c) {
+    my $method = $c->req->method;
+    $c->render(text => "You called /whatever with $method.");
+  };
+
+  app->start;
+
+=head2 Optional placeholders
+
+All placeholders require a value, but by assigning them default values you can make capturing optional.
+
+  use Mojolicious::Lite -signatures;
+
+  # /hello
+  # /hello/Sara
+  get '/hello/:name' => {name => 'Sebastian', day => 'Monday'} => sub ($c) {
+    $c->render(template => 'groovy', format => 'txt');
+  };
+
+  app->start;
+  __DATA__
+
+  @@ groovy.txt.ep
+  My name is <%= $name %> and it is <%= $day %>.
+
+Default values that don't belong to a placeholder simply get merged into the stash all the time.
+
+=head2 Restrictive placeholders
+
+A very easy way to make placeholders more restrictive are alternatives, you just make a list of possible values.
+
+  use Mojolicious::Lite -signatures;
+
+  # /test
+  # /123
+  any '/:foo' => [foo => ['test', '123']] => sub ($c) {
+    my $foo = $c->param('foo');
+    $c->render(text => "Our :foo placeholder matched $foo");
+  };
+
+  app->start;
+
+All placeholders get compiled to a regular expression internally, this process can also be customized. Just make sure
+not to use C<^> and C<$>, or capturing groups C<(...)>, non-capturing groups C<(?:...)> are fine though.
+
+  use Mojolicious::Lite -signatures;
+
+  # /1
+  # /123
+  any '/:bar' => [bar => qr/\d+/] => sub ($c) {
+    my $bar = $c->param('bar');
+    $c->render(text => "Our :bar placeholder matched $bar");
+  };
+
+  app->start;
+
+You can take a closer look at all the generated regular expressions with the command L<Mojolicious::Command::routes>.
+
+  $ ./myapp.pl routes -v
+
+=head2 Under
+
+Authentication and code shared between multiple routes can be realized easily with routes generated by
+L<Mojolicious::Lite/"under">. All following routes are only evaluated if the callback returned a true value.
+
+  use Mojolicious::Lite -signatures;
+
+  # Authenticate based on name parameter
+  under sub ($c) {
+
+    # Authenticated
+    my $name = $c->param('name') || '';
+    return 1 if $name eq 'Bender';
+
+    # Not authenticated
+    $c->render(template => 'denied');
+    return undef;
+  };
+
+  # Only reached when authenticated
+  get '/' => 'index';
+
+  app->start;
+  __DATA__
+
+  @@ denied.html.ep
+  You are not Bender, permission denied.
+
+  @@ index.html.ep
+  Hi Bender.
+
+Prefixing multiple routes is another good use for it.
+
+  use Mojolicious::Lite;
+
+  # /foo
+  under '/foo';
+
+  # /foo/bar
+  get '/bar' => {text => 'foo bar'};
+
+  # /foo/baz
+  get '/baz' => {text => 'foo baz'};
```

Comment:
Perhaps document here what the auto-generated route name of a route under an under is (i.e. "baz" versus "foobaz").